### PR TITLE
Covid-Tracker version 1

### PR DESCRIPTION
**Pull request by Ibizugbe**
In this PR I have completed the entire project and met the required specifications of the project.
- no linter error.
- the project is professionally documented.
- best practices have been utilized using JavaScript.
- I have fetched the COVID API.
- I have made use of React, and redux to build the interface.
- I have performed the required test on the project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,86 @@
...
+- `npx eslint .` : To test Eslint Report
+
+### To test the application
+
+- `npm install --save-dev jest`
+- `npm test`
+
+## Live Demo
+
+- [Visit here](https://space-travellers-97.netlify.app/)
```
Comment (zhadier) [inline README.md]: - [ ] Kindly make sure you deploy the project and add the correct deployment link in your ReadMe. Currently, the link seems to take me to the space travelers project. You can use Netlify or Heroku for deployment but I personally recommend Netlify since Heroku will abandon it's free services in september.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,86 @@
...
+
+## Show your support
+
+Give a star if you like this project!
+
+## Acknowledgments
+
+- Microverse linters
+- Coding partners
+
```
Comment (zhadier) [inline README.md]: - [ ] Kindly ensure that you acknowledge the creator of the design, Nelson Sakwa, and add a link to his Behance page as is required in his License.

File: src/__test/DisplayLists.test.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  const tree = Renderer.create(
+    <React.StrictMode>
+      <Provider store={store}>
+        <BrowserRouter>
+          <DisplayLists />
+        </BrowserRouter>
+      </Provider>
+    </React.StrictMode>,
+  ).toJSON();
+  expect(tree).toMatchSnapshot();
```
Comment (zhadier) [inline src/__test/DisplayLists.test.js:18]: - [ ] Please ensure you have more test coverage in this Project
    - Please add Unit tests for the Reducer functions (You can just stub the new State and it is very easy since reducers are pure functions by default)
    -  Kindly also add some integration tests for your project to ensure user interactivity between components is as expected. 

Tests will ensure that your project is robust and less prone to edge case bugs

**Review by zhadier — CHANGES_REQUESTED** [review]
# Hi @Ibizugbe  👋 

Good job so far! 🌤️ 
There are some issues that you still need to work on to go to the next project but you are almost there! 🍓 



## Highlights
✔️ No linter errors
✔️ GitFlow was set up correctly
✔️ Professional pull request title and description


## Required Changes ♻️

- [ ] Kindly ensure that the details page is responsive at around 400px since this is in essence a Mobile first application so mobile responsiveness is absolutely necessary. Right now due to the font-size, the rows are overflowing the container and the pink background does not take up the entire space. Kindly fix this issue

image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question @zhadier so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test/DisplayLists.test.js (hunk 3 above)
Comment (Ibizugbe) [reply src/__test/DisplayLists.test.js:18]: Thank you @zhadier for your review, I will implement all the changes required.

File: src/components/DisplayLists.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+          <div className="col-12">
+            <h3 className="text-center">Africa</h3>
+            <form className="form d-flex justify-content-center align-items-center">
+              <div className="mb-3 border px-3 py-2">
+                <input
+                  className="search"
+                  type="text"
+                  value={searchValue}
+                  placeholder="Search country"
+                  onChange={filterFunction}
```
Comment (FrederickMih) [inline src/components/DisplayLists.js:49]: - [ ] Great job implementing this filtering function :+1:  Based on the technical requirement of this project, filtering was supposed to be done in a `Filter` stateless component to filter data retrieve from the API. I'll kindly advise you to fix this issue since the previous reviewer didn't notice this.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @Ibizugbe , 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## Suggested changes

- [ ] Hi @Ibizugbe, as requested by the previous reviewer, it's a technical requirement for this project to write unit tests for several of your components and some pure functions in redux code. You ought to implement..

_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/components/DisplayLists.js (hunk 4 above)
Comment (Ibizugbe) [reply src/components/DisplayLists.js:49]: Hello, @FrederickMih  Thanks for your feedback, I appreciate your time. Please I would love it if you could throw more light on this part of the review. The search input is not a requirement in the project, It's just a code I added to make my work better. line 46 to 49 above is not where the data from the API was filtered.

Comment (Ibizugbe) [reply src/components/DisplayLists.js:49]: Thanks for the review generally @FrederickMih , I will do well to write a test for the pure functions in redux

Comment (FrederickMih) [reply src/components/DisplayLists.js:49]: Hi @Ibizugbe, it's a requirement of this project to Filter data retrieved from the API. Filtering should obviously be done on an input search field. Check out the third point on the technical requirements listing screenshot below:

Screenshot from 2022-09-03 19-11-12

**Review by mateo951 — APPROVED** [review]
### Hi @Ibizugbe, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it ✔

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by Ibizugbe** [comment]
Thanks for your time @mateo951
